x/shared/types/service.go

```diff
@@ -24,20 +22,22 @@ func init() {
 	// Compile the regex pattern
 	regexExprServiceId = regexp.MustCompile(regexServiceId)
 	regexExprServiceName = regexp.MustCompile(regexServiceName)
-
 }
 
-// IsValidService checks if the provided ServiceId struct has valid fields
-// TODO_TECHDEBT(BETA): Refactor to a `Service#ValidateBasic` method.
-func IsValidService(service *sharedtypes.Service) bool {
-	// Check if service Id and Name are valid using the provided helper functions
-	return service != nil &&
-		IsValidServiceId(service.Id) &&
-		IsValidServiceName(service.Name)
+// ValidateBasic performs basic stateless validation of a Service.
+func (s *Service) ValidateBasic() error {
+	if !IsValidServiceId(s.Id) {
+		return ErrSharedInvalidService.Wrapf("invalid service ID: %s", s.Id)
+	}
+
+	if !IsValidServiceName(s.Name) {
+		return ErrSharedInvalidService.Wrapf("invalid service name: %s", s.Name)
+	}
+
+	return nil
```

Comment:
This might require adding unit tests.

Reply:
`Service` evolved. Good catch!

Reply:
Adding these fields to the validation breaks a lot of code and tests because we're creating services with `Id` only  in a lot of places, I propose to replace these `Service`s with `Id`s as stated in `service.proto`

>   // TODO_MAINNET: Avoid embedding the full Service because we just need the ID.

I estimate it could be done in 1 day. WDYT?

Reply:
How about this:

1. Add `TODO_BETA/TODO_BLOCKER(@red-0ne): ...` in the validation function explaining this situation
2. Update the PR and merge it in
3. Iterate on the ServiceId thing next week